apps/api/src/server/services/UserService.ts

```diff
@@ -48,6 +54,92 @@ export class UserController {
     // changedRows would show actual db changes (but is deprecated)
     return { updated: results[0].affectedRows === 1 };
   }
+
+  public async getFriendshipIds(id: number, status: FriendshipStatus[] = ['accepted']) {
+    const friendshipIds = await this.resources.db
+      .select({
+        inviterId: friendships.inviterId,
+        inviteeId: friendships.inviteeId,
+      })
+      .from(friendships)
+      .where(and(
+        inArray(friendships.status, status),
+        or(eq(friendships.inviterId, id), eq(friendships.inviteeId, id)),
+      ));
+    // get ids just of the other users in the friendship
+    return friendshipIds.map(friend => (friend.inviterId === id ? friend.inviteeId : friend.inviterId));
+  }
+
+  public async getFriendProfiles(friendIds: number[]) {
+    return await this.resources.db
+      .select({
+        id: users.id,
+        firstName: users.firstName,
+        lastName: users.lastName,
+        displayName: users.displayName,
+        profileImageUrl: users.profileImageUrl,
+        email: users.email,
+        twitter: users.twitter,
+        facebook: users.facebook,
+        instagram: users.instagram,
+      })
+      .from(users)
+      .where(inArray(users.id, friendIds)) as FriendProfile[];
+  }
+
+  public async getFriendslist(id: number) {
+    const friendIds = await this.getFriendshipIds(id);
+    const friendProfiles = await this.getFriendProfiles(friendIds);
+    const friendsOfFriends = await Promise.all(friendProfiles.map(friend => this.getFriendshipIds(friend.id!)));
+
+    friendProfiles.forEach((friend, index) => {
+      const mutualFriendsCount = friendsOfFriends[index].filter(friendId => friendIds.includes(friendId)).length;
+      friend.mutualFriends = mutualFriendsCount;
+    });
+
+    return friendProfiles;
+  }
+
+  public async getFriendRequests(id: number) {
+    const pendingFriendships = await this.resources.db
+      .select({
+        inviterId: friendships.inviterId,
+        inviteeId: friendships.inviteeId,
+        message: friendships.message,
+      })
+      .from(friendships)
+      .where(and(
+        eq(friendships.status, 'pending'),
+        eq(friendships.inviteeId, id),
+      ));
```

Comment:
Would like to use `getFriendshipIds()` but this only looks for friendships where User is _invitee_.